broker/src/main/java/io/zeebe/broker/exporter/stream/ExporterDirector.java

```diff
@@ -227,7 +227,11 @@ private void onSnapshotRecovered() {
 
     clearExporterState();
 
-    actor.submit(this::readNextEvent);
+    if (state.hasExporters()) {
+      actor.submit(this::readNextEvent);
+    } else {
+      actor.close();
```

Comment:
could we close the actor sooner if we anyways know there are no exporters? or is the point of this change to do this after the state.setPosition above here at line 222?

Reply:
The point is after cleaning up the state to close the actor yes. Before we haven't done it, which caused this issue. Its about line 228.

Reply:
Thanks for explaining